src/libs/SelectionScraper/index.js

```diff
@@ -1,151 +1,3 @@
-                    } else {
-                        node = range.commonAncestorContainer.parentNode.closest(`[${tagAttribute}]`);
-                    }
-
-                    // This means "range.commonAncestorContainer" is a text node. We simply get its parent node.
-                    if (!node) {
-                        node = range.commonAncestorContainer.parentNode;
-                    }
-
-                    node = node.cloneNode();
-                    node.appendChild(clonedSelection);
-                    div.appendChild(node);
-                }
-            }
-
-            return div.innerHTML;
-        }
-
-        return window.getSelection().toString();
-    }
-
-    // If browser doesn't support Selection API, returns empty string.
-    return '';
-};
-
-/**
- * Clears all attributes from dom elements
- * @param {Object} dom htmlparser2 dom representation
- * @returns {Object} htmlparser2 dom representation
- */
-const replaceNodes = (dom) => {
-    let domName = dom.name;
-    let domChildren;
-    const domAttribs = {};
-    let data;
-
-    // Encoding HTML chars '< >' in the text, because any HTML will be removed in stripHTML method.
-    if (dom.type === 'text') {
-        data = Str.htmlEncode(dom.data);
-    }
-
-    // We are skipping elements which has html and body in data-testid, since ExpensiMark can't parse it. Also this data
-    // has no meaning for us.
-    if (dom.attribs && dom.attribs[tagAttribute]) {
-        if (!elementsWillBeSkipped.includes(dom.attribs[tagAttribute])) {
-            domName = dom.attribs[tagAttribute];
-        }
-
-        // Adding a new line after each comment here, because adding after each range is not working for chrome.
-        if (isCommentTag(dom.attribs[tagAttribute])) {
-            dom.children.push(new Element('br', {}));
-        }
-    }
-
-    // We need to preserve href attribute in order to copy links.
-    if (dom.attribs && dom.attribs.href) {
-        domAttribs.href = dom.attribs.href;
-    }
-
-    if (dom.children) {
-        domChildren = _.map(dom.children, c => replaceNodes(c));
-    }
-
-    return {
-        ...dom,
-        data,
-        name: domName,
-        attribs: domAttribs,
-        children: domChildren,
-    };
-};
-
-/**
- * Reads html of selection, replaces with proper tags used for markdown, parses to markdown.
- * @returns {String} parsed html as String
- */
-const getAsMarkdown = () => {
-    const selectionHtml = getHTMLOfSelection();
-
-    const domRepresentation = parseDocument(selectionHtml);
-    domRepresentation.children = _.map(domRepresentation.children, c => replaceNodes(c));
-
-    const newHtml = render(domRepresentation);
-
-    const parser = new ExpensiMark();
-
-    return parser.htmlToMarkdown(newHtml);
-};
-
-const SelectionScraper = {
-    getAsMarkdown,
-};
-
-export default SelectionScraper;
+export default BaseSelectionScraper;
```

Comment:
This means we can scrap BaseSelectionScraper and use the old index.js and index.native.js files.

Reply:
Now we need a base function in index.native.js.